imports/wasi_snapshot_preview1/fs.go

```diff
@@ -97,17 +97,17 @@ func fdAllocateFn(_ context.Context, mod api.Module, params []uint64) syscall.Er
 		return syscall.EINVAL
 	}
 
-	st, err := f.Stat()
-	if err != nil {
-		return platform.UnwrapOSError(err)
+	st, errno := f.Stat()
+	if errno != 0 {
+		return errno
 	}
 
 	if st.Size >= tail {
 		// We already have enough space.
 		return 0
 	}
 
-	osf, ok := f.File.(truncateFile)
+	osf, ok := f.File.File().(truncateFile)
```

Comment:
so the idea of this iterative approach shows up places like this. Any time you see `File.File()`, you know there's a function not yet implemented on the base type that we need. When all of these disappear, the file type is feature complete and the function `File.File()` can be removed.